baseplate/secrets/fetcher.py

```diff
@@ -224,10 +282,18 @@ def main():
     parser.readfp(args.config_file)
     fetcher_config = dict(parser.items("secret-fetcher"))
 
+    auth_types = {
```

Comment:
It feels strange for things outside of `VaultClientFactory` to be aware of `_vault_aws_ec2_auth`. Where might  be a better place to define auth types?